cogs/ReactForRole.py

```diff
@@ -0,0 +1,814 @@
+        :return:
+        """
+        try:
+            role: discord.Role = await commands.RoleConverter().convert(ctx, role_str)
+            await ctx.send(f"Okay, I'll add {role.name} to the list of roles required for RFR usage on the server.")
+            self.rfr_database_manager.add_guild_rfr_required_role(ctx.guild.id, role.id)
+        except (commands.CommandError, commands.BadArgument):
+            await ctx.send("Found an issue with your provided argument, couldn't get an actual role. Please try again.")
+
+    @commands.check(KoalaBot.is_admin)
+    @commands.check(rfr_is_enabled)
+    @react_for_role_group.command("removeRequiredRole")
+    async def rfr_remove_guild_required_role(self, ctx: commands.Context, role_str: str):
+        """
+        Removes a role from perms for use of rfr functionality in a server, so you can specify that you need, e.g.
+        "@Student" to be able to use rfr functionality in the server. It's server-wide permissions handling however. By
+        default anyone can use rfr functionality in the server. User needs to have admin perms to use.
+        :param ctx: Context of the command
+        :param role_str: Role ID/name/mention
+        :return:
+        """
+        try:
+            role: discord.Role = await commands.RoleConverter().convert(ctx, role_str)
+            await ctx.send(
+                f"Okay, I'll remove {role.name} from the list of roles required for RFR usage on the server.")
+            self.rfr_database_manager.remove_guild_rfr_required_role(ctx.guild.id, role.id)
+        except (commands.CommandError, commands.BadArgument):
+            await ctx.send("Found an issue with your provided argument, couldn't get an actual role. Please try again.")
+
+    @commands.check(KoalaBot.is_admin)
+    @commands.check(rfr_is_enabled)
+    @react_for_role_group.command("listRequiredRoles")
+    async def rfr_list_guild_required_roles(self, ctx: commands.Context):
+        """
+        Lists the server-specific role permissions for using rfr functionality. If list is empty, any role can use rfr
+        functionality.
+        :param ctx: Context of the command.
+        :return:
+        """
+        role_ids = self.rfr_database_manager.get_guild_rfr_required_roles(ctx.guild.id)
+        msg_str = "You will need one of these roles to react to rfr messages on this server:\n"
+        KoalaBot.logger.info(role_ids)
+        for role_id in role_ids:
+
+            role: discord.Role = discord.utils.get(ctx.guild.roles, id=role_id)
+            if not role:
+                KoalaBot.logger.error(f"couldn't find role {role_id} in guild {ctx.guild.id}. Please check.")
+            else:
+                msg_str += f"{role.mention}\n"
+        await ctx.send(msg_str)
+
+    async def prune_rfr_roles(self, guild: discord.Guild):
+        pass
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_remove(self, payload: discord.RawReactionActionEvent):
```

Comment:
Nice comments so far!
docstrings here and lower pls <3
p.s. I realise you were probs just in the middle of doing it and I decide to spring another review on you, my bad XD